### PR TITLE
Extendable data reader factory

### PR DESCRIPTION
**Pull request by alexfurmenkov**

### CODE REVIEW

File: cdisc_rules_engine/services/data_readers/data_reader_factory.py [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+class DataReaderFactory(FactoryInterface):
+    _service_map = {
+        "xpt": XPTReader,
+    }
+
+    def __init__(self):
+        self._default_service_name: str = "xpt"
 
+    @classmethod
+    def register_service(cls, name: str, service: Type[DataReaderInterface]):
```
Comment (nhaydel) [inline cdisc_rules_engine/services/data_readers/data_reader_factory.py:19]: These methods should probably be called register reader and get_reader. Also the _service_map should be reader_map

**Review by nhaydel — APPROVED** [review]
Approved with suggestions to rename methods and variable

File: cdisc_rules_engine/services/data_readers/data_reader_factory.py (hunk 1 above)
Comment (alexfurmenkov) [reply cdisc_rules_engine/services/data_readers/data_reader_factory.py:19]: We tend to make all factories implement the same interface for convenience. The interface defines the methods as register_service, get_service. Good catch about reader_map, will do.

Comment (nhaydel) [reply cdisc_rules_engine/services/data_readers/data_reader_factory.py:19]: Sounds good. I will update the operations factory to use get_service, and register_service method

Comment (alexfurmenkov) [reply cdisc_rules_engine/services/data_readers/data_reader_factory.py:19]: Yeah, just inherit them from interface

**Pull request merged**